File: src/main/java/no/fint/provider/springer/config/MongoIndexConfig.java
package no.fint.provider.springer.config;

import no.fint.provider.springer.storage.Springer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Sort;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.data.mongodb.core.index.Index;
import org.springframework.stereotype.Component;

import jakarta.annotation.PostConstruct;

@Component
public class MongoIndexConfig {

    @Autowired
    private MongoTemplate mongoTemplate;

    @PostConstruct
    public void ensureIndexes() {
        mongoTemplate.indexOps(Springer.class)
                .ensureIndex(new Index().on("type", Sort.Direction.ASC));
    }
}

File: src/main/kotlin/no/fint/provider/springer/config/MongoIndexConfig.kt
package no.fint.provider.springer.config

import jakarta.annotation.PostConstruct
import no.fint.provider.springer.storage.Springer
import org.springframework.data.domain.Sort
import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.data.mongodb.core.index.Index
import org.springframework.stereotype.Component

@Component
class MongoIndexConfig(
    private val mongoTemplate: MongoTemplate
) {
    @PostConstruct
    fun ensureIndexes() {
        mongoTemplate.indexOps(Springer::class.java)
            .createIndex(Index().on("type", Sort.Direction.ASC))
    }
}
